docs/tested-kubernetes-versions.md

```diff
@@ -0,0 +1 @@
+This page is populated for released Argo Workflows versions. Use the version selector to view this table for a specific version.
```

Comment:
Does this mean that we will only generate the table on releases? If that's the case should we manually add the matrix here until the next release is made?

Reply:
Correct, this file will be only be replaced with the table when the `Docs` workflow is run on release branches. But that doesn't make this message inaccurate, because the release branches will still have old docs containing the `Kubernetes Compatibility Matrix`  table.

Also, using the cherry-pick bot that @Joibel just added (https://github.com/argoproj/argo-workflows/pull/14151), it should be easy to cherry-pick this onto `release-3.6`.